packages/pages/src/vite-plugin/build/closeBundle/functionMetadata.ts

```diff
@@ -61,21 +60,40 @@ const getFunctionMetadataMap = async (
 async function generateFunctionMetadata(
   filepath: string
 ): Promise<[string, FunctionMetadata]> {
-  const buildResult = await esbuild.build({
-    entryPoints: [filepath],
-    outdir: TEMP_DIR,
-    write: false,
-    format: "esm",
-    bundle: true,
-  });
-  const importedFile = await importFromString(buildResult.outputFiles[0].text);
+  project.addSourceFileAtPath(filepath);
+  const defaultExportDeclaration = project
+    .getSourceFile(filepath)
+    ?.getDefaultExportSymbol()
+    ?.getDeclarations()[0];
   const relativePath = path.relative(process.cwd(), filepath);
-
-  if (!importedFile.default) {
-    throw new Error(`${relativePath} does not contain a default export.`);
+  if (Node.isExportAssignment(defaultExportDeclaration)) {
+    const exportIdenfitier = defaultExportDeclaration.getChildrenOfKind(
+      SyntaxKind.Identifier
+    )[0];
+    if (!exportIdenfitier) {
+      throw new Error(
+        `${relativePath} contains a default export assignment that is improperly ` +
+          "formatted. It may have an anonymous arrow function as the default " +
+          "export which is not supported."
+      );
+    }
+    return [relativePath, { entrypoint: exportIdenfitier.getText() }];
```

Comment:
What happens if the default export assignment is not a function?

Reply:
Unfortunately this current way does not check that the default export is a function.  I don't think that there's a way to easily check with ts-morph if something is a function or not since the exported function could be derived from something imported, in which case ts-morph has no way of knowing what it is:

```
import pluginFactory from 'some/module';
const plugin = pluginFactory("params");
export default plugin;
```

The approach we had before also did not check that the default export was a function, so I don't think any functionality is being lost.  But I do agree that a function check would be nice to have.

Reply:
I'll think some more about this!

Reply:
Got it, I'm fine with leaving it as is then